src/Metrics/ReporterFactory.h

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: BSD-2-Clause
+//
+// This code has been produced by the European Spallation Source
+// and its partner institutes under the BSD 2 Clause License.
+//
+// See LICENSE.md at the top level for license information.
+//
+// Screaming Udder!                              https://esss.se
+
+#pragma once
+
+#include <memory>
+
+namespace Metrics {
+
+class Reporter;
+class Registrar;
+enum struct LogTo;
+
+/// Create a Reporter with a specified type of Sink
+std::unique_ptr<Reporter>
+createReporter(std::shared_ptr<Registrar> const &MetricsRegistrar,
```

Comment:
I do not see a good reason to have this factory.

Reply:
It is intended to tidy away creation of the `MetricsList` which is then shared between the `Registrar` and the `Reporter`. It is not really factory pattern, so should at least be renamed.